### ISSUE TITLE
Evaluate getting rid of gulp and npm

### ISSUE THREAD

**Issue by chucker** [Blazored/Gitter#81]
Best as I can see, we currently use the gulp (and npm) infrastructure for two reasons:

* it compiles SCSS to CSS, does some further jiggling, and copies the result to the two target projects
* it copies the JS* to the two target projects

*) it does _not_ compile TS to JS: the MSBuild toolchain does that instead.

This works, but it's kind of an ugly and brittle fit. We might want to move to things like `BuildWebCompiler`.

**Comment by SQL-MisterMagoo** [comment]
We used to use (Build)WebCompiler but it has issues with @import when it is combining files (it doesn't move them to the top of the file) - and it is no longer maintained I believe. I did fork it and fix that to see how easy it would be (really very) but never submitted a PR because it's not being maintained - and I don't want to maintain it either!

**Comment by SQL-MisterMagoo** [comment]
It might be worth revisiting though because the build time is terrible with gulp

**Comment by chucker** [comment]
> We used to use (Build)WebCompiler but [..] it is no longer maintained I believe. 

Yeah, seems that way. I'm kind of hoping that Blazor will lead to a renaissance. Importing the npm ecosystem seems kind of ridiculous in this context.

> the build time is terrible with gulp

I may have made it a lot slower with the change to `npx` (which first runs npm to check if the version of gulp is current before executing it, but does come with the advantage that you don't have to manually install anything).

**Comment by chucker** [comment]
https://github.com/excubo-ag/WebCompiler

> This project is based on madskristensen/WebCompiler. However, the dependency to node and the node modules have been removed, to facilitate a pure dotnet core implementation.
